core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -416,6 +416,9 @@ public void updateRoutingEntry(ShardRouting newRouting) throws IOException {
                 // active primaries.
                 throw new IndexShardRelocatedException(shardId(), "Shard is marked as relocated, cannot safely move to state " + newRouting.state());
             }
+            assert newRouting.active() == false || state == IndexShardState.STARTED || state == IndexShardState.RELOCATED ||
```

Comment:
to make sure I don't miss something: this PR doesn't make this assertion before, it was relevant before as well, right?